integration/yarn/src/test/java/alluxio/yarn/ClientTest.java

```diff
@@ -0,0 +1,162 @@
+    client.parseArgs(new String[] {
+        "-resource_path", "test",
+        "-am_memory", Integer.toString(appMasterMem),
+        "-am_vcores", "2"});
+    client.run();
+  }
+
+  @Test(timeout = 1000)
+  public void notEnoughVCoreForApplicationMaster() throws Exception {
+    int appMasterMem = 1024;
+    int appMasterCore = 2;
+    Resource resource = Resource.newInstance(appMasterMem, appMasterCore / 2);
+    generateMaxAllocation(resource);
+    mThrown.expect(RuntimeException.class);
+    mThrown.expectMessage(ExceptionMessage.YARN_NOT_ENOUGH_RESOURCES.getMessage(
+        "ApplicationMaster", "virtual cores", appMasterCore, resource.getVirtualCores()));
+    Client client = new Client();
+    client.parseArgs(new String[] {
+        "-resource_path", "test",
+        "-am_memory", Integer.toString(appMasterMem),
+        "-am_vcores", Integer.toString(appMasterCore)});
+    client.run();
+  }
+
+  @Test(timeout = 10000)
+  public void notEnoughMemoryForAlluxioMaster() throws Exception {
+    int masterMemInMB = (int) (Configuration.getBytes(
+        PropertyKey.INTEGRATION_MASTER_RESOURCE_MEM) / Constants.MB);
+    Resource resource = Resource.newInstance(masterMemInMB / 2, 4);
+    generateMaxAllocation(resource);
+    mThrown.expect(RuntimeException.class);
+    mThrown.expectMessage(ExceptionMessage.YARN_NOT_ENOUGH_RESOURCES.getMessage(
+        "Alluxio Master", "memory", masterMemInMB, resource.getMemory()));
+    Client client = new Client();
+    client.run();
+  }
+
+  @Test(timeout = 10000)
+  public void notEnoughVCoreForAlluxioMaster() throws Exception {
+    int masterMemInMB = (int) (Configuration.getBytes(
+        PropertyKey.INTEGRATION_MASTER_RESOURCE_MEM) / Constants.MB);
+    int masterVCores = Configuration.getInt(PropertyKey.INTEGRATION_MASTER_RESOURCE_CPU);
+    Resource resource = Resource.newInstance(masterMemInMB, 3);
+    generateMaxAllocation(resource);
+    mThrown.expect(RuntimeException.class);
+    mThrown.expectMessage(ExceptionMessage.YARN_NOT_ENOUGH_RESOURCES.getMessage(
+        "Alluxio Master", "virtual cores", masterVCores, resource.getVirtualCores()));
+    Client client = new Client();
+    client.run();
+  }
+
+  @Test(timeout = 10000)
+  public void notEnoughMemoryForAlluxioWorker() throws Exception {
+    int workerMemInMB = (int) (Configuration.getBytes(
+        PropertyKey.INTEGRATION_WORKER_RESOURCE_MEM) / Constants.MB);
+    int ramdiskMemInMB = (int) (Configuration.getBytes(
+        PropertyKey.WORKER_MEMORY_SIZE) / Constants.MB);
+    Resource resource = Resource.newInstance((workerMemInMB + ramdiskMemInMB) / 2, 4);
+    generateMaxAllocation(resource);
+    mThrown.expect(RuntimeException.class);
+    mThrown.expectMessage(ExceptionMessage.YARN_NOT_ENOUGH_RESOURCES.getMessage(
+        "Alluxio Worker", "memory", (workerMemInMB + ramdiskMemInMB), resource.getMemory()));
+    Client client = new Client();
+    client.run();
+  }
+
+  @Test(timeout = 10000)
```

Comment:
We don't care about performance or expect the test to time out, so we can remove the timeouts from the `@Test` annotations.